assets/javascripts/discourse/components/modal/diff-modal.gjs

```diff
@@ -25,30 +86,46 @@ export default class ModalDiffModal extends Component {
       @closeModal={{@closeModal}}
     >
       <:body>
-        {{#if @model.diff}}
-          {{htmlSafe @model.diff}}
-        {{else}}
-          <div class="composer-ai-helper-modal__old-value">
-            {{@model.oldValue}}
+        {{#if this.loading}}
+          <div class="composer-ai-helper-modal__loading">
+            <CookText @rawText={{this.selectedText}} />
           </div>
+        {{else}}
+          {{#if this.diff}}
+            {{htmlSafe this.diff}}
+          {{else}}
+            <div class="composer-ai-helper-modal__old-value">
+              {{@model.oldValue}}
+            </div>
 
-          <div class="composer-ai-helper-modal__new-value">
-            {{@model.newValue}}
-          </div>
+            <div class="composer-ai-helper-modal__new-value">
+              {{@model.newValue}}
+            </div>
+          {{/if}}
         {{/if}}
       </:body>
 
       <:footer>
-        <DButton
-          class="btn-primary confirm"
-          @action={{this.triggerConfirmChanges}}
-          @label="discourse_ai.ai_helper.context_menu.confirm"
-        />
-        <DButton
-          class="btn-flat revert"
-          @action={{this.triggerRevertChanges}}
-          @label="discourse_ai.ai_helper.context_menu.revert"
-        />
+        {{#if this.loading}}
+          <DButton
+            class="btn-primary"
+            @label="discourse_ai.ai_helper.context_menu.loading"
+            @disabled={{true}}
+          />
+        {{else}}
+          <DButton
+            class="btn-primary confirm"
+            @action={{this.triggerConfirmChanges}}
+            @label="discourse_ai.ai_helper.context_menu.confirm"
+          />
```

Comment:
Up to you but you don't necessarily need to duplicate the confirmation button, you could do something like this:

```hbs
<DButton
  class="btn-primary confirm"
  @action={{this.triggerConfirmChanges}}
  @label={{if this.loading "discourse_ai.ai_helper.context_menu.loading" "discourse_ai.ai_helper.context_menu.confirm"}}
  @disabled={{this.loading}}
/>
```

Reply:
hmmm yeah it can work, I wanted to leave the modal though as is as much as possible for the other consumers, leaving this for now